basestar-storage/src/test/java/io/basestar/storage/TestStorage.java

```diff
@@ -509,11 +509,38 @@ protected void testDelete() {
 
         final Map<String, Object> current = storage.get(Consistency.ATOMIC, schema, id, ImmutableSet.of()).join();
         assertNull(current);
-        // FIXME
-//        if(storage.storageTraits(schema).getHistoryConsistency().isStronger(Consistency.EVENTUAL)) {
-//            final Map<String, Object> v1 = storage.readObjectVersion(schema, id, 1L).join();
-//            assertNull(v1);
-//        }
+    }
+
+    @Test
+    protected void testDeleteRecreate() {
+
+        assumeTrue(supportsDelete());
+
+        final Storage storage = storage(namespace);
+
+        final ObjectSchema schema = namespace.requireObjectSchema(SIMPLE);
+
+        final String id = UUID.randomUUID().toString();
+
+        final Instance init = instance(schema, id, 1L);
+
+        storage.write(Consistency.ATOMIC, Versioning.CHECKED)
+                .createObject(schema, id, init)
+                .write().join();
+
+        storage.write(Consistency.ATOMIC, Versioning.CHECKED)
+                .deleteObject(schema, id, init)
+                .write().join();
+
+        //Test same object can be created again
+
+        storage.write(Consistency.ATOMIC, Versioning.CHECKED)
+                .createObject(schema, id, init)
+                .write().join();
```

Comment:
is this test for delete+recreate for the _same_ object version?

Reply:
It tests the current behaviour of the storage interface, which does not continue versioning from the point of delete but instead starts from 1 again and begins overwriting history - we can change this with usual process, but for now this test is mostly to catch storage implementations that would fail to overwrite the history for whatever reason.